pydicom/config.py

```diff
@@ -152,15 +153,92 @@ def DS_decimal(use_Decimal_boolean: bool = True) -> None:
 Default ``False``.
 """
 
+
 enforce_valid_values = False
-"""Raise exceptions if any value is not allowed by DICOM Standard.
+"""Obsolete.
+Use :attr:`Settings.reading_validation_mode` instead.
+"""
 
-e.g. DS strings that are longer than 16 characters; IS strings outside
-the allowed range.
 
-Default ``False``.
+# Constants used to define how data element values shall be validated
+NO_VALIDATION = 0
+"""If one of the validation modes is set this value, no value validation 
+will be performed.
+"""
+
+WARN_ON_ERROR = 1
+"""If one of the validation modes is set this value, a warning is issued fn 
+a value validation error occurs.
+"""
+
+RAISE_ON_ERROR = 2
+"""If one of the validation modes is set this value, an exception is raised 
+if a value validation error occurs.
 """
 
+
+class Settings:
+    """Collection of several configuration values.
+    Accessed via the singleton :attr:`settings`.
+
+    Attributes
+    ----------
+    reading_validation_mode : int
+        Defines behavior for value validation while reading value.
+        Value validation checks if a value is allowed by the DICOM Standard,
+        e.g. that DS strings are not longer than 16 characters and contain only
+        allowed characters.
+        The default (:attr:`WARN_ON_ERROR`) is to log a warning in the case of
+        an invalid value, :attr:`RAISE_ON_ERROR` will raise an error in this
+        case, and :attr:`NO_VALIDATION` will bypass the
+        validation (with the exception of some encoding errors).
+    writing_validation_mode : int
+        Defines behavior for value validation while writing a value.
+        See :attr:`Settings.reading_validation_mode`.
+
+    New in version 2.3
```

Comment:
For some reason, `.. versionadd:: 2.3` did not work here - this is just a workaround.

Reply:
Just a minor comment while I skim, [this should be](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionadded):
```
.. versionadded: 2.3
```

Reply:
Yes, this is what I tried (typo in the comment), and it did not work for some reason.

Reply:
Turned out the line was just in the wrong place - it has be placed before the attribute documentation, fixed now.